fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/Main.java

```diff
@@ -287,32 +298,28 @@ protected void processDB2() throws Exception {
         JdbcPropertyAdapter adapter = new Db2PropertyAdapter(configProps);
         Db2Translator translator = new Db2Translator();
         JdbcConnectionProvider cp = new JdbcConnectionProvider(translator, adapter);
+        PoolConnectionProvider connectionPool = new PoolConnectionProvider(cp, this.threads);
+        ITransactionProvider transactionProvider = new SimpleTransactionProvider(connectionPool);
 
         // Provide the credentials we need for accessing a multi-tenant schema (if enabled)
         // Must set this BEFORE we create our persistence object
-        if (this.tenantName != null) {
-            if (tenantKey == null) {
-                throw new IllegalArgumentException("No tenant-key provided");
-            }
-
-            // Set up the FHIRRequestContext on this thread so that the persistence layer
-            // can configure itself for this tenant
-            FHIRRequestContext rc = FHIRRequestContext.get();
-            rc.setTenantId(this.tenantName);
-            rc.setTenantKey(this.tenantKey);
+        if (this.tenantName == null || tenantKey == null) {
+            throw new IllegalArgumentException("No tenant-key or tenant-name provided");
```

Comment:
the message makes it sound like neither one was provided.  i suggest splitting into two different if blocks so that we can have targeted error messages.  alternatively you could reword it as "Either tenant-key or tenant-name was not provided".

Reply:
done